radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -41,26 +85,85 @@ public PendingVotes(Hasher hasher) {
 
 	/**
 	 * Inserts a vote for a given vertex, attempting to form a quorum certificate for that vertex.
-	 *
+	 * <p>
 	 * A QC will only be formed if permitted by the {@link ValidatorSet}.
+	 *
 	 * @param vote The vote to be inserted
 	 * @return The generated QC, if any
 	 */
 	public Optional<QuorumCertificate> insertVote(Vote vote, ValidatorSet validatorSet) {
-		final Hash voteHash = hasher.hash(vote.getVoteData());
-
-		ECDSASignature signature = vote.getSignature().orElseThrow(() -> new IllegalArgumentException("vote is missing signature"));
-		ValidationState validationState = votes.computeIfAbsent(voteHash, validatorSet::newValidationState);
+		final ECPublicKey voteAuthor = vote.getAuthor();
+		final VoteData voteData = vote.getVoteData();
+		final Hash voteHash = this.hasher.hash(voteData);
+		final ECDSASignature signature = vote.getSignature().orElseThrow(() -> new IllegalArgumentException("vote is missing signature"));
+		// Only process for valid validators and signatures
+		if (validatorSet.containsKey(voteAuthor)) {

```

Comment:
Could clean up the indent here by inverting this if and returning immediately if the vote author is not in the validator set

Reply:
Actually, could reduce most of the nesting that way once the above if is inverted